### ISSUE TITLE
Pour info : détection script Travian Task Queue for Travian 4

### ISSUE THREAD

**Issue by bowler26** [adipiciu/Travian-scripts#4]
Bonjour,
j'ai utilisé ce script pour lancer des constructions, il a été détecté par Travian.

Je suis sous Windows 10 et connexion via une box en France.

**Comment by adipiciu** [comment]
Hi, do you used it only for building constructions?

From what I know, travian can ban you if the account has suspicious activity. For example if it's on 24/24 hours, or if you schedule many server requests per minute it can flag the account as a automatic bot.

**Comment by Vikos2021** [comment]
Witam, u mnie też skrypt został wykryty mimo, że nie użytkowałem go 24/24 :(

**Comment by adipiciu** [comment]
Maybe they implemented some protection in the latest updates? What functions do you used? Only building constructions or something else?

**Comment by Vikos2021** [comment]
I used it mostly for farming and a few times for building constructions .

**Comment by bowler26** [comment]
indeed, I only used it to mount hiding places, memory 2 times 6 levels.
I received a 25% penalty on the entire account.
So you advise to use it with spacing in time if you have several levels to do.
I mean 1 stash level at 2:00:00 pm then a 2nd level at 2:30:00 pm, for example and so on

**Comment by adipiciu** [comment]
There should be timeframes between the server requests. So leave at least 1-2 minutes between every request. If there are too many server requests per minute, travian can flag the account as a bot.

**Comment by Qussayyon** [comment]
> Bonjour, j'ai utilisé ce script pour lancer des constructions, il a été détecté par Travian.
> 
> Je suis sous Windows 10 et connexion via une box en France.

Hi @adipiciu ,
I also got banned, I have been using the scripts RB++, TTQ for sometimes, I'm actually afraid to give more info because maybe travian have someone here looking at ur scripts issues  to relate to those who got banned in their game!

BTW:
I did try an online bot for sometimes and I did not get banned!

The only thing I can relate is: I used wave builder and I attacked one village of Natars, then after 24 hours, it's saying I violate the rules of the games, all this while I did not use wave builder, it's first time and in 24hours I got banned, maybe they are using some AI and analyse the way of players playing the game? And this happened while the plus and gold were activated.

So yeah, may account is banned, and I might stop playing the game, and that's better for me.

If you need any more info, let me know, I registered 2 new servers few days ago to try out latest features.

Kind regards.

**Comment by Qussayyon** [comment]
P.S

I just remember:
I'm using an addon from Firefox to reload the page every 5 minutes, maybe that can be a reason?

**Comment by adipiciu** [comment]
> P.S
> 
> I just remember: I'm using an addon from Firefox to reload the page every 5 minutes, maybe that can be a reason?

Yes, that's possible. Every action that is repeating at a regular interval can be detection as a script/bot.

**Comment by Qussayyon** [comment]
@adipiciu 
I would like to update you:
it seems that it's not from reload extension, but rather from elephant scanner.
I used your elephant scanner, then it got detected, and after I asked them they said the reason is:
_they determined that the ban was correct and stated that  a checking bot has been used on the IP and IP* between at least 4th and 7th November         If you have any further questions on this please let me know and I will do what i can to answer them_

**Comment by Qussayyon** [comment]
So, I think it's good idea to separate the elephant scanner and make it different script, and make for it a warning that it will send additional requests to server.

**Comment by Qussayyon** [comment]
> > travian resource bar plus was unable to use features and was also removed from Greasy Fork.I and my friends have been looking for so long to find you here >.< . What happened bro?
> 
> The script was reported by someone on greasyfork, because it's a "copy". The owner of greasyfork was retarded and deleted my script and kept the old script clone from 2014. Short story: I moved to github.

I think you should have a repo in gitlab as well.

**Comment by geses78** [comment]
slm kolay gelsin daha sonra geliştir ve daha sonra asker gönder düğmeleri çalışmıyor ilgilenireniz sevinirim

**Comment by geses78** [comment]
hello, then develop soldiers and then develop farms. can't be done. it gives an error, can you update please?

**Comment by geses78** [comment]
> @adipiciu Sizi güncellemek istiyorum: Görünüşe göre bu yeniden yükleme uzantısından değil, fil tarayıcıdan. Fil tarayıcınızı kullandım, sonra algılandı ve onlara sorduğumda sebebini söylediler: _yasağın doğru olduğunu belirlediler ve IP ve IP* üzerinde en az 4 ile 7 arasında bir kontrol botu kullanıldığını belirttiler. Kasım Bununla ilgili başka sorularınız varsa lütfen bana bildirin, onlara cevap vermek için elimden geleni yapacağım._

hello, then develop soldiers and then develop farms. can't be done. it gives an error, can you update please?

**Comment by adipiciu** [comment]
> would like to update you:
What server are you playing? 
Make a new issue next time, don't write in different off topic subject.

**Comment by Qussayyon** [comment]
Hi @adipiciu,

I want to inform you.
I have been banned 2 more times after first time, I'm not sure why!
maybe because I'm using more than one computer.
first time they said:
checking bot. second and third time they did not give reason and they did not give time period.

**Comment by geses78** [comment]
https://ts30.x3.europe.travian.com/ suvucusunda oynuyorum

Qussayyon ***@***.***>, 29 Kas 2021 Pzt, 07:42 tarihinde şunu
yazdı:

**Comment by adipiciu** [comment]
> suvucusunda oynuyorum

I don't play on that server. It was updated on 25 November. Maybe they changed something. I will try to update the script today.
Please, next time open a new topic, don't use an existing unrelated topic.

**Comment by adipiciu** [comment]
> Hi @adipiciu,
> 
> I want to inform you. I have been banned 2 more times after first time, I'm not sure why! maybe because I'm using more than one computer. first time they said: checking bot. second and third time they did not give reason and they did not give time period.

I use multiple computers and never been banned. It's very strange your case. Are you sure it's not your auto reload script? If you don't use the crop or oasis elephant scan or the attack detector which are making server requests, they can't see the script.

**Comment by Qussayyon** [comment]
@adipiciu,

I'll double check, but I think maybe it's related to sending troops in wave, even though I put delay 900ms

**Comment by adipiciu** [comment]
> hello, then develop soldiers and then develop farms. can't be done. it gives an error, can you update please?

It works for me on the international server. Contact me on skype, ID: adipiciu so we can get on Team Viewer to see what is your problem.

**Comment by adipiciu** [comment]
> hello, then develop soldiers and then develop farms. can't be done. it gives an error, can you update please?

@geses78 Hi, I have updated TTQ script to version 1.9.1. Can you confirm if it is working ok now?

**Comment by geses78** [comment]
> > merhaba, sonra askerler geliştirin ve ardından çiftlikler geliştirin. yapılamaz. hata veriyor lütfen günceller misiniz?
> 
> @geses78 Merhaba, TTQ betiğini 1.9.1 sürümüne güncelledim. Şimdi çalışıp çalışmadığını onaylayabilir misiniz?

hata devam ediyor.sonra asker bas çalışmıyor.

**Comment by geses78** [comment]
sonra bina geliştir çalışıyor fakat sonra asker bas çalışmıyor.

**Comment by adipiciu** [comment]
I fixed a bug in TTQ in version 1.9.1 where it would fail to detect the tribe/troops on Firefox. 
@bowler26 @Vikos2021 Can you confirm if you are using Firefox?

**Comment by geses78** [comment]
firefox kullanmıyorum ben chrome kullanıyorum

adipiciu ***@***.***>, 5 Ara 2021 Paz, 10:00 tarihinde şunu
yazdı:

**Comment by gojko33** [comment]
I got banned - not shure why. But i find out from some friend that they can track cursor movement with simple javascript code. 
So when you have lot of activity they check that first. Also i used wave script - probably this caused my ban. 
Just not sure about resource bar - i used only resoursce info with colours.

**Comment by geses78** [comment]
merhaba kolay gelsin çiflik işlemi çalışıyor bina geliştirme çalışıyor fakat daha sonra asker basma çalışmıyor. daha sonra askerleri göndere çalışmıyor bir güncelleme yaparmısınız.

**Comment by totinaster** [comment]
Hi, I use elephant finder and we got ban on gos.2x 🫣

**Comment by Qussayyon** [comment]
> Hi, I use elephant finder and we got ban on gos.2x 🫣

definitely.

**Comment by totinaster** [comment]
> > Hi, I use elephant finder and we got ban on gos.2x 🫣
> 
> definitely.

so i have to fix my comment as we got banned because of something else, but anyway, I think the finder isn't quite safe to use
